### PR TITLE
Manual merge of Hannes Bartosiks PTC updates from the master branch

### PR DESCRIPTION
**Pull request by HaroonRafique**
Updated new PTC (updated by Piotr Skowronski) with acceleration fix implemented by Hannes Bartosik in the master branch. Untested.

### CODE REVIEW

**Comment by HaroonRafique** [comment]
This pull request is designed to fix the following issue:

PTC_Compatibility_Issue

This occurs when trying to run the current main (master) branch of PTC-PyORBIT with a PTC flat file generated in a MAD-X version greater than 5.02.00. In this example with the latest MAD-X 5.06.01.

**Comment by HaroonRafique** [comment]
Long term tracking test (5E4 turns) for the Proton Synchrotron (pre-LIU parameters) have been performed using:

- NewPTC  GNU Fortran build (local)
- OldPTC GNU Fortran build (local)
- NewPTC intel Fortran build (on CERN AFS)
- OldPTC intel Fortran build (on CERN AFS)

PTC_PS_5E4_Turns_iFort

Results:
- Particles initially populated at < 10 sigma differ by ~ 1E-7 when comparing Intel/GNU fortran builds using the same PTC version
- Particles initially populated at < 10 sigma differ by ~ 1E-7 when comparing the new and old PTC versions
- At 5 sigma the difference is 2.67E-4 % 
- At 10 sigma the difference is 1E-3 %
- Visual agreement in Poincare section (stable particle orbits in all cases)

Tests:
https://github.com/HaroonRafique/PTC_PyORBIT_Tests/tree/main/00_NewPTC_Test/PS_Long_Term_Tracking

Plots found in this notebook:
https://github.com/HaroonRafique/PTC_PyORBIT_Tests/blob/main/00_NewPTC_Test/PS_Long_Term_Tracking/Compare_iFort_GFortran.ipynb

Or here:
https://github.com/HaroonRafique/PTC_PyORBIT_Tests/tree/main/00_NewPTC_Test/PS_Long_Term_Tracking/Plots

**Comment by HaroonRafique** [comment]
Simulations of the Proton Synchrotron (pre-LS2 1.4 GeV) Injection Bump have been performed. These simulations are benchmarked against measurements, the original results are found here: 
- https://github.com/HaroonRafique/PyORBIT_MD4224 (original full study)
- https://github.com/HaroonRafique/MD4224_Light (condensed and updated study)

These simulations do the following:
- Run MAD-X to generate ramp tables for BSW dipoles over 100 arbitrary steps as representative of the 500 turn injection bump
- Convert ramp tables to appropriate PTC format
- Run PyORBIT and match initial distribution to the first PTC Twiss table (1.ptc)
- Simulate the effect of both dipolar and sextupolar component of the BSW magnet ramp for 500 turns (from maximum bump to closure) and then for a further 1700 turns (total 2200 turns or 5 ms at 1.4 GeV)
- No Space Charge to assess PTC tracking only

The simulations have been run on CERN's HPC-Batch system.

Overall results show good agreement between PTC versions, the full results are given:
- https://github.com/HaroonRafique/PTC_PyORBIT_Tests/tree/main/00_NewPTC_Test/PS_Injection_Bump
- https://nbviewer.jupyter.org/github/HaroonRafique/PTC_PyORBIT_Tests/blob/main/00_NewPTC_Test/PS_Injection_Bump/Output_playground_PS_newPTC.ipynb

In summary the horizontal emittance behaviour is shown (note 2 working points of (Q_x, Q_y) = (6.21 ,6.24), and (6.07, 6.24): 
epsn_x_growth_cf

Conclusion:
- PTC merge complete and successful

**Comment by HaroonRafique** [comment]
After discussion with Hannes Bartosik, an additional test is required to check acceleration.

**Comment by HaroonRafique** [comment]
PSB Acceleration tests now complete. These simulations have been run on STFC SCARF. Small differences within tolerance due to difference in initial flat file between MAD-X v 5.02.00 and MAD-X v 6.02.01. 

- Simulations can be found here: https://github.com/HaroonRafique/PTC_PyORBIT_Tests/tree/main/00_NewPTC_Test/PSB/Acceleration_Test
- Notebook with output plots: https://github.com/HaroonRafique/PTC_PyORBIT_Tests/blob/main/00_NewPTC_Test/PSB/Acceleration_Test/PSB_Acceleration_Output.ipynb

## Differences:
- Closed orbit x < 1E-16 mm
- Closed orbit y < 4E-187 mm
- Mean x +/- 2E-3 mm
- Mean y +/- 2E-3 mm
- Mean z +/- 0.8 mm
- Mean xp +/- 4E-5 mrad
- Mean yp +/- 4E-5 mrad
- Beta < 1E-8
- Gamma < 1E-8
- Kinetic energy < 1E-5 MeV
- Longitudinal emittance < 0.01 eVs
- Horizontal emittance < 1.5E-8 mm mrad
- Vertical emittance < 3.5E-8 mm mrad

Overall results show good agreement between the merged PTC and original PTC versions of PyORBIT.
PyORBIT Acceleration Test_kinetic_energy
